src/pages/Swap/index.tsx

```diff
@@ -194,16 +225,34 @@ export default function Swap({ className }: { className?: string }) {
   // for expert mode
   const [isExpertMode] = useExpertModeManager()
   // swap state
-  const [state, dispatch] = useReducer(swapReducer, initialSwapState)
+  const [state, dispatch] = useReducer(swapReducer, { ...initialSwapState, ...prefilledState })
   const { typedValue, recipient, independentField } = state
+
+  const previousConnectedChainId = usePrevious(connectedChainId)
+  useEffect(() => {
+    const combinedInitialState = { ...initialSwapState, ...prefilledState }
+    console.log({ combinedInitialState })
+    if (previousConnectedChainId && previousConnectedChainId !== connectedChainId) {
+      dispatch(
+        replaceSwapState({
+          ...initialSwapState,
+          ...prefilledState,
+          field: combinedInitialState.independentField ?? Field.INPUT,
+          inputCurrencyId: combinedInitialState.INPUT.currencyId ?? undefined,
+          outputCurrencyId: combinedInitialState.OUTPUT.currencyId ?? undefined,
```

Comment:
same here - is this needed? also don't you want to clear and reset to the new chain's native token

Reply:
same reason - the reducer action expects `inputCurrencyId` and `outputCurrencyId` instead of `[Field.INPUT]: { currencyId }`

and, `combinedInitialState` at this point has the tokens we want to default to, with the new chain's native token.